src/pages/jobs.mdx

```diff
@@ -0,0 +1,111 @@
+export const description = 'Jobs'
+
+# Jobs
+
+Nitric provides functionality for provisioning long running batch workloads and High Performance Computing (HPC).
+
+## Definitions
+
+### Batch
+
+A Batch is like a Nitric service, but is designed to be run as a single unit of work with a start and a finish.
+
+### Job Definitions
+
+Job definitions are defined as a part of batches and represent a single unit of work that work that can be run within a Nitric `Batch`
+
+### Job
+
+A Job is an instance of a Job Definition that is running within a `Batch`, these can be created from Nitric services or other Nitric batches.
+
+## Limitations of Batches
+
+Jobs are designed to be long running HPC workloads and can take some time to spin up. They are not designed with reactivity in mind and are not suitable for responding to events from cloud resources.
+
+Jobs are unable to run the following:
+
+- Topic Subscriptions
+- Bucket Notifications
+- API & HTTP resources
+- Websocket message handlers
+
+Jobs can be used to read and write to/from all nitric resources.
+
+## Defining Batches
+
+Batches are defined in a similar way to services, within the `nitric.yaml` file. As an example
+
+```yaml
+batches:
+  - match: ./batches/*.ts
+    start: yarn dev:batches $BATCH_PATH
+```
+
+> A Batch can contain any number of Job Definitions.
+
+## Defining a Job
+
+Within a Batch we define Job Definitions, which describes the code we will execute in each job.
+
+<CodeGroup>
+
+```typescript
+import { jobDefinition, JobContext } from '@nitric/sdk'
+
+const myJob = jobDefinition('analyse').define((ctx: JobContext) => {
+  // Do some work
+})
+```
+
+```python
+from nitric.resources import job_definition
+from nitric.application import Nitric
+from nitric.context import JobContext
+
+# Define will allow the developer to fine-tune the job option, such as adding resource dependencies (e.g. CPU, Memory, GPUs) etc.
+@job_definition("analyse").define()
+async def generate_image(ctx: None):
+    # Do some work
+
+
+Nitric.run()
+```
+
+</CodeGroup>
+
+## Submitting Jobs for Execution
+
+Jobs may be submitted from Nitric `services`.
+
+<CodeGroup>
+
+```javascript
+import * as nitric from '@nitric/sdk'
+
+const api = nitric.api('public')
+const analyseJob = nitric.jobDefintion('analyse')
+
+api.post('/submit-job', async (ctx) => {
+  await analyseJob.submit({
+    someKey: 'someValue',
+  })
```

Comment:
We should also be able to define hardware requirements here as additional options:
```typescript
await analyseJob.submit({
  someKey: 'someValue',
}, {
   cpus: 1,
   memory: 1024,
   gpus: 1,
})
```

Reply:
Should the compute requirements be specified in the job definition rather than the submission?

```js
const analyseJob = nitric.jobDefintion('analyse', {
   cpus: 1,
   memory: 1024,
   gpus: 1,
})
```

Maybe defaults can be specified on the job definition, but overrides are available to the user at the job submission?

Reply:
Thought about that, Azure and GCP don't have any where to natively store this information, so we'd need to shim in a config store (e.g. buckets or env variables) to hold this info so we can access it at runtime.

The simplest approach would be to allow runtime only, users could set their own defaults using env variables, and we could implement default configuration as a new feature in the future.

Reply:
Thinking about this a bit further we'll still need to store the `image` reference somewhere accessible at runtime as well, so it might not hurt to include defaults along with this config as well.

Reply:
Have taken this into consideration and make the `Jobs` part of the deployment spec a `message` rather than a string to this detail can be included.